vllm/model_executor/models/mixtral.py

```diff
@@ -114,12 +121,23 @@ def __init__(
                 self.num_total_experts, device="cuda", dtype=torch.float32),
             requires_grad=False) if self.use_fp8 else None
 
-        set_weight_attrs(self.ws, {
-            "weight_loader": self.weight_loader,
-        })
-        set_weight_attrs(self.w2s, {
-            "weight_loader": self.weight_loader,
-        })
+        # Scaling factors for FP8 activations
+        need_act_scales = (self.use_fp8 and
+                           linear_method.quant_config.act_scaling == "static")
+        self.as_scale = nn.Parameter(
+            torch.zeros(1, device="cuda", dtype=torch.float32),
```

Comment:
nit: Theoretically, we shouldn't use "cuda" in the model code. Since the GPU worker sets "cuda" as the default device in torch, device="cuda" is not necessary. Also, it's not good for the compatibility with non-CUDA devices.

This rule is violated for Mixtral and other MoE models unfortunately. 😢

Reply:
Ok, I'll make a follow up PR to remove the `device="cuda"` -- since we also specify it explicitly for the other parameters, I don't want to be inconsistent for this PR :)